tests/phpunit/elementor/includes/managers/test-controls.php

```diff
@@ -54,4 +57,61 @@ public function test_delete_stack() {
 		// Assert
 		$this->assertNull( Plugin::$instance->controls_manager->get_stacks( $widget->get_unique_name() ) );
 	}
+
+	public function test_clear_stack_cache() {
+		// Arrange
+		$stacks_mock = [
+			'button' => [
+				'controls' => [
+					'border_border' => [
+					'type'=>'select',
+					'tab'=>'style',
+					'section'=>'section_style',
+					'label'=>'Border Type',
+					'options'=>[''=>'Default'],
+					'none'=>'None',
+					'solid'=>'Solid',
+					'double'=>'Double',
+					'dotted'=>'Dotted',
+					'dashed'=>'Dashed',
+					'groove'=>'Groove',
+					'selectors'=>['{{WRAPPER}} .elementor-button'=>'border-style: {{VALUE}};'],
+					'separator'=>'before',
+					'classes'=>'elementor-group-control-border elementor-group-control elementor-group-control-border',
+					'name'=>'border_border',
+					'default'=>''
+					],
+				],
+			],
+		];
+
+		$widget = new Mock_Widget( [
+			'settings' => [],
+			'id' => '1',
+		], $stacks_mock );
+
+		// Act
+		Plugin::$instance->controls_manager->open_stack( $widget );
+
+		// Assert
+		$stacks = Plugin::$instance->controls_manager->get_stacks();
+		$this->assertNotEmpty( $stacks );
```

Comment:
Elementor is instantiated for the test run, so theoretically there could be unrelated stacks in the cache. Did you check that your mock widget's stack is the only one? If it isn't the only stack this could be a false negative

Reply:
I checked if the current `stacks` are actually our stacks. 
They seems to be OK, and not false positive. 

I also found some issue with the assigned the control_data to the actual widget.
`add_control_to_stack` that was fixed.